checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,315 @@
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hadoop-MapReduce-on-Alluxio.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  private Path mOutputFilePath;
+  private FileSystem mFileSystem;
+
+  /**
+   * An empty input format that can generate input splits to affect the number of mapper nodes.
+   */
+  static class EmptyInputFormat extends InputFormat<Object, Object> {
+    public static boolean sCreateDone = false;
+
+    /**
+     * An empty input split.
+     */
+    static class EmptyInputSplit extends InputSplit implements Writable {
+
+      public EmptyInputSplit() { }
+
+      @Override
+      public long getLength() throws IOException {
+        return 0;
+      }
+
+      @Override
+      public String[] getLocations() throws IOException {
+        return new String[]{};
+      }
+
+      @Override
+      public void readFields(DataInput in) throws IOException {
+      }
+
+      @Override
+      public void write(DataOutput out) throws IOException {
+      }
+    }
+
+    /**
+     * A record reader converting input to record-oriented view.
+     */
+    static class EmptyRecordReader extends RecordReader<Object, Object> {
+
+      public EmptyRecordReader() {
+      }
+
+      @Override
+      public void initialize(InputSplit split, TaskAttemptContext context) {
+      }
+
+      @Override
+      public void close() {
+      }
+
+      @Override
+      public Object getCurrentKey() {
+        return new Object();
+      }
+
+      @Override
+      public Object getCurrentValue() {
+        return new Object();
+      }
+
+      @Override
+      public float getProgress() {
+        if (sCreateDone) {
+          return 1.0f;
+        } else {
+          return 0.0f;
+        }
+      }
+
+      @Override
+      public boolean nextKeyValue() {
+        if (sCreateDone) {
+          return false;
+        } else {
+          sCreateDone = true;
+          return true;
+        }
+      }
+    }
+
+    @Override
+    public RecordReader<Object, Object> createRecordReader(InputSplit split,
+        TaskAttemptContext context) {
+      return new EmptyRecordReader();
+    }
+
+    /**
+     * Creates the desired number of splits.
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) {
+      int numSplits = job.getConfiguration().getInt(MRJobConfig.NUM_MAPS, 1);
+      List<InputSplit> splits = new ArrayList<>();
+      for (int split = 0; split < numSplits; split++) {
+        splits.add(new EmptyInputSplit());
+      }
+      return splits;
+    }
+  }
+
+  /**
+   * In each mapper node, we will check whether this node can recognize
+   * Alluxio classes and filesystem.
+   */
+  protected static class FooMapper extends Mapper<Object, Object, Text, Text> {
```

Comment:
we should not call this `FooMapper` :D 
Foo means something. in here you can put `Foo=Checker` and call this mapper `CheckerMapper`, reducer `CheckerReducer`

Reply:
hahhhhh, I misunderstand :D